pystan/plots/plots_data.py

```diff
@@ -0,0 +1,332 @@
+        # reshape and translate for plotting
+        vecs = vecs.reshape([vecs.shape[0]*vecs.shape[1]]+list(par_dims), order='F').T
+        if split_pars:
+            limits = None
+        else:
+            limits = vecs.min(), vecs.max()
+        m = np.multiply.reduce(par_dims)
+        indices = np.c_[np.unravel_index(np.arange(m), par_dims, order='F')]
+        for idx in indices:
+            vec = vecs[tuple(idx)]
+            # use par if value scalar
+            if par_dims == (1,):
+                name = par
+            else:
+                # add 1 for the index
+                name = "{}[{}]".format(par, ",".join(map(str, idx+1)))
+
+            if isinstance(vec[0], (int, np.integer, np.uint)):
+                hist, edges = hist_data(vec, limits, nbins=nbins)
+                yield {'par' : par,
+                       'name' : name,
+                       'vec' : vec,
+                       'hist' : (hist/hist.max()*overlap, edges)}
+            else:
+                x_kde, y_kde = kde_data(vec, limits, c=c)
+                yield {'par': par,
+                       'name' : name,
+                       'vec' : vec,
+                       'kde' : (x_kde, y_kde/y_kde.max()*overlap)}
+
+def parcoords_data(fit, pars, divergence=False, **kwargs):
+    """Function to gather data for the parcoords-plot.
+
+    Parameters
+    ----------
+    fit : StanFit4Model instance
+    pars : tuple
+        parameters plotted
+    divergence : bool, optional
+        If True, return divergent sample independently
+    transform : str, function
+        If str, use {'minmax', 'standard'}
+            minmax = (arr - min) / (max - min)
+            standard = (arr - mean) / std
+        If function, transform data with tranform-function.
+        E.g. function=np.log, arr = np.log(arr)
+    inc_warmup : bool, optional
+
+    Returns
+    -------
+    data : list
+        List contains = (names, data, divergent data (if divergence is True))
+    """
+    inc_warmup = kwargs.pop('inc_warmup', False)
+    vectors = []
+    names = []
+    transform = kwargs.pop('transform', None)
+    for par in pars:
+        vecs = fit.extract(pars=par, permuted=False, inc_warmup=inc_warmup)[par]
+        par_dims = vecs.shape[2:]
+        if not par_dims:
+            vecs = np.expand_dims(vecs, -1)
+            par_dims = vecs.shape[2:]
+        m = np.multiply.reduce(par_dims)
+        vecs = vecs.reshape([vecs.shape[0]*vecs.shape[1]]+list(par_dims), order='F')
+        vecs = vecs.reshape(-1, m, order='F')
+        if transform is not None:
+            if isinstance(transform, str):
+                if transform.lower() in ('min-max', 'minmax'):
+                    vmin, vmax = vecs.min(0), vecs.max(0)
+                    vecs = (vecs - vmin) / (vmax-vmin)
+                elif transform.lower() in ('standard'):
+                    vmean, vstd = vecs.mean(0), vecs.std(0, ddof=1)
+                    vecs = (vecs - vmean) / vstd
```

Comment:
Add inverse cdf normalization. 

I have an approximation for it working, but I want it to work correctly. Currently the problem is that without some clever summing the cumsum will move unique numbers to different values

    # Current situation
    1, 2, 2, 3 --> 1, 3, 5, 8

    # Option 1 (max value)
    1, 2, 2, 3 --> 1, 5, 5, 8

    # Option 2 (mean value)
    1, 2, 2, 3 --> 1, 4, 4, 8

    # Option 3 (min value)
    1, 2, 2, 3 --> 1, 3, 3, 8

I prefer the mean one, notice however that the cumsum is going over the block normally.